vernac/declare.ml

```diff
@@ -1646,17 +1646,25 @@ let rec_tac_initializer finite guard thms snl =
        | (id,n,_)::l -> Tactics.mutual_fix id n l 0
        | _ -> assert false
 
-let start_with_initialization ~info ~cinfo sigma =
+let process_using env evd ~fixnames ~terms using =
+  Proof_using.definition_using env evd ~fixnames ~terms ~using
+
+let start_with_initialization ~info ~cinfo ?using sigma =
   let { CInfo.name; typ; args } = cinfo in
   let init_tac = Tactics.auto_intros_tac args in
   let pinfo = Proof_info.make ~cinfo:[cinfo] ~info () in
-  let lemma = start_proof_core ~name ~typ:(EConstr.of_constr typ) ~pinfo ?sign:None sigma in
+  (* XXX XXX using *)
```

Comment:
What are this "XXX XXX" and the other one reminding?

Reply:
Ah, I see, it is probably about the value to give to `~terms`.

In #18742, I used `let terms = List.map EConstr.of_constr (fixtypes @ List.map_filter (fun x -> x) fixdefs) in` (where the fixdefs and fixtypes are in the `CInfo.t`).

Reply:
Done, indeed the using interpretation is now happening in "sealed" from in `declare.ml`, things look good to me.

Reply:
In line 768-771 of  current declare.ml in the PR, don't you want to compute a unique `using` for all constants of the block, computing it with all `typ` and `body` together? I have an example which otherwise would fail. (See the tests I added in `test-suite/success/proof_using.v` and `test-suite/success/definition_using.v` in #18742. I propose you add these tests to your PR too.)

Reply:
And idem in the current version of `add_mutual_definitions` in the PR.

Reply:
Thanks, will rebase on top of your tests!

Reply:
And indeed fix this todo

Reply:
Should be fixed.